ZWave is not working after 1.4 upgrade

After upgrade to version 1.4 ZWave adapter cannot start.
Here is what I see in the log:
host.iobroker  2019-05-06 17:18:06.721  warn  instance system.adapter.zwave.0 terminated due to SIGABRT
Caught  2019-05-06 17:18:06.721  error  by controller[1]: what(): Manager.cpp:1768 - InvalidValueIDError (101) Msg: Invalid ValueID passed to GetValueLabel
Caught  2019-05-06 17:18:06.720  error  by controller[1]: terminate called after throwing an instance of 'OpenZWave::OZWException'
Caught  2019-05-06 17:18:06.719  error  by controller[0]: Unhandled OpenZWave notification: 31
zwave.0  2019-05-06 17:18:06.481  info  State created: zwave.0.NODE6.THERMOSTAT_MODE.Mode_1 = undefined, index = 0, comClass = 64, instance = 1
Also I noticed that   /usr/local/lib/libopenzwave.so.1.6  is added to the lib directory. But it looks like this new library is not used.
pi@iobroker:~ $ ldd -r /opt/iobroker/node_modules/openzwave-shared/build/Release                                                                                                                                                             /openzwave_shared.node
 says libopenzwave.so.1.4 => /usr/local/lib/libopenzwave.so.1.4 

Thanks


 

 libopenzwave.so.1.4 => /usr/local/lib/libopenzwave.so.1.4 (0x76e04000)

I have the Same Problem. So I've tried to install the Version 1.3.1 via 'npm install iobroker.zwave@1.3.1 --unsafe-perm --production --save --prefix "/opt/iobroker"' but this crashes with 'npm ERR! Failed at the iobroker.zwave@1.3.1 preinstall script.'
[NPM.log](https://github.com/ioBroker/ioBroker.zwave/files/3157097/NPM.log)

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -4,8 +4,9 @@ environment:
     - nodejs_version: '6'
     - nodejs_version: '8'
     - nodejs_version: '10'
-    - nodejs_version: '12'
-    - nodejs_version: ''
+    # Node 12 is blocked by https://github.com/OpenZWave/node-openzwave-shared/issues/280
+    # - nodejs_version: '12'
+    # - nodejs_version: ''
 platform:
   - x86
   - x64

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -23,6 +23,6 @@ test_script:
   # Package tests fail for now because materialize is not supported
   # - npm run test:package
   - npm run test:unit
-  - export DEBUG=testing:*
+  - set DEBUG=testing:*
   - npm run test:integration
 build: 'off'

--- a/lib/preinstall.js
+++ b/lib/preinstall.js
@@ -140,10 +140,16 @@ function installOpenZwave() {
 }
 
 function installOpenZwaveFromSource() {
-  doScript('curl -L -O https://github.com/OpenZWave/open-zwave/archive/master.zip');
-  doScript('rm -rf open-zwave-master');
-  doScript('unzip master.zip && rm master.zip');
-  // 2019-05-03: PR #1125 has been merged today, which causes the command to fail.
-  // doScript('cd open-zwave-master && curl -O -L https://github.com/OpenZWave/open-zwave/pull/1125.patch && patch -p1 < 1125.patch');
-  doScript(`cd open-zwave-master && make && ${isRoot ? '' : 'sudo '}make install`);
+  // For now we manually have to install 1.4 because openzwave-shared is not compatible yet
+  const branch = '1.4';
+  doScript(`curl -L -O https://github.com/OpenZWave/open-zwave/archive/${branch}.zip`);
+  doScript(`rm -rf open-zwave-${branch}`);
+  doScript(`unzip ${branch}.zip && rm ${branch}.zip`);
+  try {
+  doScript(`cd open-zwave-${branch} && curl -O -L https://github.com/OpenZWave/open-zwave/pull/1125.patch && patch -p1 < 1125.patch`);
+  } catch (e) {
+    // Depending on which branch we target, PR #1125 might have been merged already,
+    // so this command will fail. No biggie though.
+  }
+  doScript(`cd open-zwave-${branch} && make && ${isRoot ? '' : 'sudo '}make install`);
 }

--- a/package.json
+++ b/package.json
@@ -40,7 +40,7 @@
     "url": "git+https://github.com/ioBroker/ioBroker.zwave.git"
   },
   "dependencies": {
-    "openzwave-shared": "^1.4.8",
+    "openzwave-shared": "AlCalzone/node-openzwave-shared#ozw-1.4",
     "@iobroker/adapter-core": "^1.0.3"
   },
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -40,7 +40,7 @@
     "url": "git+https://github.com/ioBroker/ioBroker.zwave.git"
   },
   "dependencies": {
-    "openzwave-shared": "^1.4.3",
+    "openzwave-shared": "^1.4.8",
     "@iobroker/adapter-core": "^1.0.3"
   },
   "devDependencies": {

